hw1/seq_transformer.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Sep 27 18:39:56 2022
+
+@author: anja1
+
+Программа, которая в бесконечном цикле считывает команды от
+пользователя. После команды программа запрашивает у пользователя
+последовательность нуклеиновой кислоты и распечатывает результат.
+
+Список команд:
+    exit — завершение исполнения программы
+    transcribe — напечатать транскрибированную последовательность
+    reverse — напечатать перевёрнутую последовательность
+    complement — напечатать комплементарную последовательность
+    reverse complement — напечатать обратную комплементарную последовательность
+
+
+
+"""
+
+# Словари для транскибции и комплиментации
+dic_tr = {'A':'U','T':'A','G':'C', 'C':'G',
+          'a':'u','t':'a','g':'c','c':'g'}
+
+dic_com_DNA = {'A':'T','T':'A','G':'C', 'C':'G',
+               'a':'t','t':'a','g':'c','c':'g'}
+
+dic_com_RNA = {'A':'U','U':'A','G':'C', 'C':'G',
+               'a':'u','u':'a','g':'c','c':'g'}
+
+# Список разрешенных символов на вхождение
+allow = ['a','u','t','g','c','A','T','G','C','U']
+
+# Функции для работы команд
+def reverse(seq):
+    print(seq[::-1])
+
+def transcribe(seq):
+    tbl = seq.maketrans(dic_tr)
+    print(seq.translate(tbl))
+    
+def complement(seq):
+    if 'u' in seq.lower():
+        tbl = seq.maketrans(dic_com_RNA)
+    else:
+        tbl = seq.maketrans(dic_com_DNA)
+    print(seq.translate(tbl))
+    
+def rev_complement(seq):
+    if 'u' in seq.lower():
+        tbl = seq.maketrans(dic_com_RNA)
+    else:
+        tbl = seq.maketrans(dic_com_DNA)
+    print(seq.translate(tbl)[::-1])
+
+# Бесконечный цикл для работы программы
+while 0 == 0:
+    
+    # Ввод команды пользователем
+    command = input('Enter command:')
+    
+    # Выход из программы при команде 'Exit'
+    if command == 'exit':
+        print('Good luck!')
+        break
+    
+    # Ввод последовательности пользователем
+    seq = input('Enter sequence:')
+    
+    # Переменные для проверяющего цикла
+    a = 0
+    b = 0
+    c = 0
```

Comment:
Плохие имена для переменных. Для флагов стоит использовать `True` и `False`, это более читаемо
```suggestion
    correct_alphabet_flag = True
    correct_nucleic_acid_type_flag = True
    transcription_allowed_flag = True
```